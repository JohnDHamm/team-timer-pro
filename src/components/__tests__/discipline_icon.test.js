import React from 'react'
import renderer from 'react-test-renderer'
import DisciplineIcon from '../discipline_icon'

describe('<DisciplineIcon />', () => {
  const iconStyle = { width: 20, tintColor: 'green' };

  test('it renders correctly for swim', () => {
    const tree = renderer.create(<DisciplineIcon disc={'swim'} iconStyle={iconStyle} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders correctly for bike', () => {
    const tree = renderer.create(<DisciplineIcon disc={'bike'} iconStyle={iconStyle} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders correctly for run', () => {
    const tree = renderer.create(<DisciplineIcon disc={'run'} iconStyle={iconStyle} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it does not render anything for invalid discipline', () => {
    const tree = renderer.create(<DisciplineIcon disc={'walk'} iconStyle={iconStyle} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
