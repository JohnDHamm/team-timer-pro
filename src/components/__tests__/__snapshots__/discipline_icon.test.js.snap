// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisciplineIcon /> it does not render anything for invalid discipline 1`] = `null`;

exports[`<DisciplineIcon /> it renders correctly for bike 1`] = `
<Image
  source={1}
  style={
    Object {
      "height": 15.267175572519083,
      "tintColor": "green",
      "width": 20,
    }
  }
/>
`;

exports[`<DisciplineIcon /> it renders correctly for run 1`] = `
<Image
  source={1}
  style={
    Object {
      "height": 16.129032258064516,
      "tintColor": "green",
      "width": 20,
    }
  }
/>
`;

exports[`<DisciplineIcon /> it renders correctly for swim 1`] = `
<Image
  source={1}
  style={
    Object {
      "height": 13.468013468013467,
      "tintColor": "green",
      "width": 20,
    }
  }
/>
`;
